### PR TITLE
Add table aws_vpc_eip_address_transfer closes #1519

### LINKED ISSUE #1519: Add table aws_vpc_eip_address_transfer
**Is your feature request related to a problem? Please describe.**
AWS released a feature on EIPs to transfer addresses to other accounts.  
https://aws.amazon.com/about-aws/whats-new/2022/10/amazon-virtual-private-cloud-vpc-transfer-elastic-ip-addresses-between-aws-accounts/

Would be good to extend the aws_vpc_eip table to include these additional attributes to identify EIPs being enabled for transfer.

Additional columns could be the ones defined in the describe-address-transfers:
https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-address-transfers.html

e.g. 

- **TransferAccountId -> (string)**. The ID of the account that you want to transfer the Elastic IP address to.
- **TransferOfferExpirationTimestamp -> (timestamp)**. The timestamp when the Elastic IP address transfer expired. When the source account starts the transfer, the transfer account has seven hours to allocate the Elastic IP address to complete the transfer, or the Elastic IP address will return to its original owner.
- **TransferOfferAcceptedTimestamp -> (timestamp)**.  The timestamp when the Elastic IP address transfer was accepted.
- **AddressTransferStatus -> (string)**.  The Elastic IP address transfer status.

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_vpc_eip_address_transfer
+----------------------------+-------------------------+-------------+---------------------+-----------------------------------+-------------------------------------+----------------------------+-----------+-----------+--------------+---------------------------+
| allocation_id              | address_transfer_status | public_ip   | transfer_account_id | transfer_offer_accepted_timestamp | transfer_offer_expiration_timestamp | title                      | partition | region    | account_id   | _ctx                      |
+----------------------------+-------------------------+-------------+---------------------+-----------------------------------+-------------------------------------+----------------------------+-----------+-----------+--------------+---------------------------+
| eipalloc-0602d016851ad5923 | pending                 | 52.45.46.39 | 123589368987        | <null>                            | 2022-12-30T16:36:50+05:30           | eipalloc-0602d016851ad5923 | aws       | us-east-1 | 128602159384 | {"connection_name":"aws"} |
+----------------------------+-------------------------+-------------+---------------------+-----------------------------------+-------------------------------------+----------------------------+-----------+-----------+--------------+---------------------------+

Time: 6.0s. Rows fetched: 1. Hydrate calls: 1.
> select
  allocation_id,
  address_transfer_status,
  public_ip,
  transfer_account_id,
  transfer_offer_accepted_timestamp
from
  aws_vpc_eip_address_transfer
where
  transfer_offer_accepted_timestamp >= now() - interval '30' day;
+---------------+-------------------------+-----------+---------------------+-----------------------------------+
| allocation_id | address_transfer_status | public_ip | transfer_account_id | transfer_offer_accepted_timestamp |
+---------------+-------------------------+-----------+---------------------+-----------------------------------+
+---------------+-------------------------+-----------+---------------------+-----------------------------------+

Time: 12ms. Rows fetched: 1 (cached). Hydrate calls: 0.
> select
  allocation_id,
  address_transfer_status,
  public_ip,
  transfer_account_id,
  transfer_offer_expiration_timestamp
from
  aws_vpc_eip_address_transfer
where
  transfer_offer_expiration_timestamp >= now() - interval '10' day;
+----------------------------+-------------------------+-------------+---------------------+-------------------------------------+
| allocation_id              | address_transfer_status | public_ip   | transfer_account_id | transfer_offer_expiration_timestamp |
+----------------------------+-------------------------+-------------+---------------------+-------------------------------------+
| eipalloc-0602d016851ad5923 | pending                 | 52.45.46.39 | 123793682495        | 2022-12-30T16:36:50+05:30           |
+----------------------------+-------------------------+-------------+---------------------+-------------------------------------+
```
</details>

### CODE REVIEW

**Comment by rajlearner17** [comment]
@ParthaI Pls re-work on this to map the new changes in 0.93.0

File: aws/table_aws_vpc_eip_address_transfer.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	}
+}
+
+//// LIST FUNCTION
+
+func listVpcEipAddressTransfers(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	address := h.Item.(types.Address)
+	allocationId := d.KeyColumnQualString("allocation_id")
+
+	if allocationId != "" {
```
Comment (bigdatasourav) [inline aws/table_aws_vpc_eip_address_transfer.go:75]: Please add a comment here as to why we are checking this condition

**Comment by bigdatasourav** [comment]
@ParthaI Please check the suggested changes and rebase your branch with the main.

**Review by karanpopat — APPROVED** [review]
LGTM

File: aws/table_aws_vpc_eip_address_transfer.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "address_transfer_status",
+				Description: "The Elastic IP address transfer status.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "public_ip",
+				Description: "The Elastic IP address being transferred.",
```
Comment (misraved) [inline aws/table_aws_vpc_eip_address_transfer.go:41]: Is the description correct?

File: aws/table_aws_vpc_eip_address_transfer.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	}
+}
+
+//// LIST FUNCTION
+
+func listVpcEipAddressTransfers(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	address := h.Item.(types.Address)
+	allocationId := d.EqualsQualString("allocation_id")
+
+	// Avoid api call if user specified allocation_id doesn't match the hydrate id.
```
Comment (misraved) [inline aws/table_aws_vpc_eip_address_transfer.go:77]:
```suggestion
	// Avoid API call if user-specified allocation_id doesn't match the hydrate ID from hydrate data.
```

File: docs/tables/aws_vpc_eip_address_transfer.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Table: aws_vpc_eip_address_transfer
+
+An Elastic IP Address Transfer is a static, public IPv4 address that transfer from AWS account (source account) to any other AWS account in the same AWS Region.
```
Comment (misraved) [inline docs/tables/aws_vpc_eip_address_transfer.md:3]:
```suggestion
An Elastic IP Address Transfer is a static, public IPv4 address that transfers from an AWS account (source account) to any other AWS account in the same AWS Region.
```

File: aws/table_aws_vpc_eip_address_transfer.go
Comment (misraved) [inline aws/table_aws_vpc_eip_address_transfer.go:1]: @ParthaI Is there a way by which the users can join the data of this table with the other VPC tables?

Can we add an example query that shows all the EIastic IP transfers for a VPC?

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments. Thanks!!

File: aws/table_aws_vpc_eip_address_transfer.go (hunk 2 above)
Comment (ParthaI) [reply aws/table_aws_vpc_eip_address_transfer.go:41]: The same description has mentioned in the API response. https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AddressTransfer.html

**Review by misraved — APPROVED** [review]

**Pull request merged**